bitcoinspv/clients/sui/sui_spv_client.go

```diff
@@ -2,44 +2,49 @@ package sui
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"strconv"
 
-	"github.com/block-vision/sui-go-sdk/models"
-	"github.com/block-vision/sui-go-sdk/signer"
-	"github.com/block-vision/sui-go-sdk/sui"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/wire"
+	"github.com/fardream/go-bcs/bcs"
 	"github.com/gonative-cc/relayer/bitcoinspv/clients"
 	"github.com/gonative-cc/relayer/bitcoinspv/types"
+	"github.com/pattonkan/sui-go/sui"
+	"github.com/pattonkan/sui-go/sui/suiptb"
+	"github.com/pattonkan/sui-go/suiclient"
+	"github.com/pattonkan/sui-go/suisigner"
 	"github.com/rs/zerolog"
 )
 
 const (
 	insertHeadersFunc = "insert_headers"
 	containsBlockFunc = "exist"
-	getChainTipFunc   = "latest_block_hash"
+	headFunc          = "head"
 	verifySPVFunc     = "verify_tx"
 	lcModule          = "light_client"
-	defaultGasBudget  = "10000000000"
+	// TODO: Use better defaultGasBudget
+	defaultGasBudget = 10000000000
 )
 
-// SPVClient implements the BitcoinSPV interface, interacting with a
-// Bitcoin SPV light client deployed as a smart contract on Sui.
-type SPVClient struct {
-	logger      zerolog.Logger
-	suiClient   *sui.Client
-	signer      *signer.Signer
-	lcObjectID  string
-	lcPackageID string
+// BTCLightClientObject implements the BitcoinSPV interface, interacting with a
+// Bitcoin SPV light client deployed as a smart contract on Sui.logger
```

Comment:
```suggestion
// Bitcoin SPV light client deployed as a smart contract on Sui.
```

Reply:
why not keep the old name? I am not sure about the name including the `Object` in it